### PR TITLE
Add enums for Months and other small cleanup

### PR DESCRIPTION
**Pull request by arigilder**
Adds an enum for Months and Holiday Types, instead of using integers to refer to them, to avoid confusion. A few other small cleanup/comments.

### CODE REVIEW

File: hdate/htables.py [hunk 1]
```diff
@@ -165,131 +181,192 @@ def move_if_not_on_dow(original, replacement, dow_not_orig, dow_replacement):
...
+    HOLIDAY(HolidayTypes.HOLOCAUST_MEMORIAL, "yom_hashoah", 
+            ([26, 27, 28], Months.Nisan), "",
             [move_if_not_on_dow(27, 28, 6, 0) or
              move_if_not_on_dow(27, 26, 4, 3),
              year_is_after(5718)],
             LANG(u"Yom HaShoah", DESC(u"יום השואה", u"יום השואה"))),
-    HOLIDAY(8, "yom_hazikaron", ([2, 3, 4], 8), "",
+    HOLIDAY(HolidayTypes.HOLOCAUST_MEMORIAL, "yom_hazikaron", 
+            ([2, 3, 4], Months.Iyyar), "",
             [year_is_after(5708), year_is_before(5764),
```
Comment (tsvi) [inline hdate/htables.py]: Please rename this,. Yom hazikaron is the memorial day for fallen soldiers. Maybe MEMORIAL_DAY?

File: hdate/htables.py [hunk 2]
```diff
@@ -165,131 +181,192 @@ def move_if_not_on_dow(original, replacement, dow_not_orig, dow_replacement):
...
+            (16, Months.Nisan), "DIASPORA", [],
             LANG(u"Pesach II", DESC(u"שני של פסח", u"ב' פסח"))),
-    HOLIDAY(9, "family_day", (30, 5), "ISRAEL", [],
+    HOLIDAY(HolidayTypes.ISRAEL_ONLY_MODERN_HOLIDAY, "family_day", 
+            (30, Months.Shvat), "ISRAEL", [],
             LANG(u"Family Day", DESC(u"יום המשפחה", u"יום המשפחה"))),
-    HOLIDAY(9, "memorial_day_unknown", (7, [6, 14]), "ISRAEL", [],
+    HOLIDAY(HolidayTypes.ISRAEL_ONLY_MODERN_HOLIDAY, "memorial_day_unknown", 
+            (7, [Months.Adar, Months.Adar_II]), "ISRAEL", [],
             LANG(u"Memorial day for fallen whose place of burial is unknown",
```
Comment (tsvi) [inline hdate/htables.py]: Maybe this should also be memorial day?

File: hdate/htables.py [hunk 3]
```diff
@@ -165,131 +181,192 @@ def move_if_not_on_dow(original, replacement, dow_not_orig, dow_replacement):
...
                  DESC(u"יום זכרון...", u"יום זכרון..."))),
-    HOLIDAY(9, "rabin_memorial_day", ([11, 12], 2), "ISRAEL",
+    HOLIDAY(HolidayTypes.ISRAEL_ONLY_MODERN_HOLIDAY, "rabin_memorial_day", 
+            ([11, 12], Months.Marcheshvan), "ISRAEL",
             [move_if_not_on_dow(12, 11, 4, 3), year_is_after(5757)],
```
Comment (tsvi) [inline hdate/htables.py]: Same here

**Comment by arigilder** [comment]
Changed to memorial day, switched the remaining family day to "Israel national holiday"

**Pull request merged**